### PR TITLE
[ADD][11.0]project_wbs_task

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
This module integrates tasks in the wbs structure to make easier the management of task when using wbs. It adds those features:
- The possibility to search for task by the WBS complete reference or name.

- A button in the project tree view that will conduct the user to the list view for the associated tasks.
Depends on
- [x] project_wbs https://github.com/OCA/project/pull/416 Do not merge until oca_dependency.txt explicit branch to that PR is removed

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
when accesing runbot I got `502 Bad Gateway` all the time :face_with_head_bandage:

**Comment by pedrobaeza** [comment]
It seems a problem after upgrading to Debian Jessie the servers. Let's see if @gurneyalex can take a look when coming back from holidays. Any way, it seems rebuilding a second time makes the trick.

**Comment by AaronHForgeFlow** [comment]
I'll try to rebuild. Thanks @pedrobaeza

**Comment by AaronHForgeFlow** [comment]
When runbot is ready:
image

Is there a way to force rebuild without closing and opening the PR again or committing again? (committing does not work as first build never succeed on runbot, shows `502 Bad Gateway`)

**Review by tarteo — COMMENTED** [review]
Isn't it better to make this part of project_wbs?

**Comment by AaronHForgeFlow** [comment]
@tarteo For me it's much easier to put both modules in the same Pull Request but the convention is to open separate Pull Request for separate modules, isn't it?

**Comment by tarteo** [comment]
@aheficent I mean to put the functionalities provided by this module into project_wbs. So there's only one module.

**Comment by AaronHForgeFlow** [comment]
I think it's better to keep it separated. We can manage a company with projects and do not use tasks. In that case we will have some functionality that is not really used.

**Review by tarteo — APPROVED** [review]
@aheficent Sounds right. :+1:

File: project_wbs_task/model/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
```
Comment (nikul-serpentcs) [inline project_wbs_task/model/__init__.py]: @aheficent IMHO remove  copyright from all  __init__.py

**Comment by AaronHForgeFlow** [comment]
Hi @nikul-serpentcs  could you please update your review? Thanks

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

**Comment by bennals** [comment]
@aheficent Still getting exactly the same error I reported in #416 after updating project_wbs_task with latest commit (3 days ago).  

`ERROR: insert or update on table "project_task" violates foreign key constraint "project_task_parent_id_fkey"
DETAIL:  Key (parent_id)=(3512) is not present in table "project_task".`

Restarted the server, updated both wbs modules, still no change.  Simply refreshing the browser still removes the error on subsequent attempts to duplicate the project though, so it's not a show-stopper.

Let me know if I can help in any way.

**Comment by AaronHForgeFlow** [comment]
Hi @bennals I am not able to reproduce in my local. Are you trying to duplicate another project? or does that happen when opening the previous project with that issue?

It seems to me that because of the old code there are some bad references in your database. I would try to recreate the links between the tasks and the analytic accounts applying a query like the following:

> UPDATE project_task pt set analytic_account_id = aa.id
FROM account_analytic_account aa
INNER JOIN project_project pp ON pp.analytic_account_id = aa.id
WHERE pt.project_id = pp.id

**Review by rafaelbn — CHANGES_REQUESTED** [review]
This module is WIP as the dependency is in PR with bots red :red_circle: 

https://github.com/OCA/project/pull/417/files#diff-e2ace23e896fd3bf8c371c93bb1f6e43R16

Please if you want to help review https://github.com/OCA/project/pull/416#issuecomment-525806156

**Comment by AaronHForgeFlow** [comment]
Hi @rafaelbn CI are green now. Could you update your review?

**Comment by AaronHForgeFlow** [comment]
cc @ForgeFlow

**Review by MiquelRForgeFlow — APPROVED** [review]

File: project_wbs_task/README.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* A button in the project tree view that will conduct the user to the list
+  view for the associated tasks.
+* The possibility to search for task by the WBS complete reference or name.
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/87/10.0
```
Comment (MiquelRForgeFlow) [inline project_wbs_task/README.rst]:
```suggestion
   :target: https://runbot.odoo-community.org/runbot/87/11.0
```

**Comment by AaronHForgeFlow** [comment]
Just squashed my commits

**Comment by AaronHForgeFlow** [comment]
Rebased to take the fix done in #416

**Comment by AaronHForgeFlow** [comment]
As long the only pending review was due to failing CI as now that is fixed, this is ready to be merged. Will merge soon.

**Comment by AaronHForgeFlow** [comment]
Reopening to see if runbot fetches the latest code in base. The failing runbot was due to an old issue resolved already in 11.0 branch.
It seems its not rebuilding :(

**Comment by AaronHForgeFlow** [comment]
Superseded by https://github.com/OCA/project/pull/655

**Comment by AaronHForgeFlow** [comment]
coronapr_small

**Pull request closed without merging**